File: three-sum/ThreeNumberSum.swift
import Foundation

class ThreeNumberSum {
    func threeNumbersSum(array: inout [Int], targetSum: Int) -> [[Int]] {
        array.sort()
        var results: [[Int]] = []
        for index in stride(from: 0, to: array.count, by: 1) {
            var left = index + 1
            var right = array.count - 1
            while(left < right) {
                var sum = array[index] + array[left] + array[right]
                if(sum == targetSum) {
                    results.append([ array[index], array[left], array[right] ])
                    left += 1; right -= 1;
                } else if (sum < targetSum) {
                    left += 1
                } else {
                    right -= 1
                }
            }
        }
        return results
    }
}

File: three-sum/ThreeNumberSum.cs
using System;

public class ThreeNumberSums {
    public static List<int[]> ThreeNumbersSum(int[] array, int targetSum) {
        List<int[]> results = new List<int[]>();
        Array.Sort(array);
        for(int i = 0; i < array.Length; i++){
            int left = i  + 1;
            int right = array.Length - 1;
            while(left < right) {
                int sum = array[i] + array[left] + array[right];
                if(sum.Equals(targetSum)) {
                    results.Add(new int[] { array[i], array[left], array[right] });
                    left++; right--;
                } else if (sum < targetSum) {
                    left++;
                } else {
                    right--;
                }
            }
        }
        return results;
    }
}